### ISSUE TITLE
(Feature req.) more advanced time view

### ISSUE THREAD

**Issue by Wjxfi** [tahaak67/Farhan#16]
I came from a more advanced application of application time control. There I can see how much time I've spent on all my apps in a week, month, 2, 3, 4, 4, 5, 6, 7, 8, 9, 10, 11 months, a year, a year and a half, two years

And, in addition to simple numbers, it shows which applications I was in for how many minutes.

Also, then the app sends notifications like "this week you looked at your phone for 4 hours less!" Or "this week you looked at your phone for 9 more hours!"

I would like to be able to view this information in this application as well.

The application I'm talking about is https://play.google.com/store/apps/details?id=org.coober.myappstime
(it has trackers)

**Comment by tahaak67** [comment]
This is interesting, i tried the application you mentioned above and it reports wrong usage amount some times.
In Android (As far as i know) the only ways to get usage data are:

- Get usage events, filter that data and calculate the duration manually (Farhan does this)
This way is the most accurate but as reported in #20  its going to be slow (sometimes very slow) because it involves **getting** all usage events for that day **and** filtering when the app was visible/invisible **and** sum everything for every package alone **and then** convert milliseconds to a readable duration like '1h10m'.
**Also** when querying data using this method only a limited number of days is provided by Android so it's not really possible to get usage for a duration that's older than a week.

- Get the information directly using the Android API
The app you mentioned is most likely using this method and that's probably why it reports inaccurate data because that's a [known issue](https://issuetracker.google.com/issues/199494936) [[1](https://issuetracker.google.com/issues/118564471)] for a long time in the Android APIs and there isn't really anything obvious about whether we are going to have a fix for it or not

**In short:** i will keep this feature in mind in case i find a workaround, other than that this will take a while unfortunately :( 
Thanks for the suggestion!

**Comment by tahaak67** [comment]
I'll need more time to look into that in depth i have to make sure that's going to be possible without making more weird bugs

**Comment by tahaak67** [comment]
the latest commit in [0.1.2/usage_fix](https://github.com/tahaak67/Farhan/tree/0.1.2/usage_fix) have the base implementation to this feature and also a couple of other fixes such as #20 it might have some bugs for now (as far i have seen minor UI bugs) so if someone is interested they can build the project from source and try it to provide feedback before release.

**Comment by tahaak67** [comment]
Hi again got not update from you so i hope you are fine
I'm going to close this issue since its fixed in the release.
feel free to comment if you have feedback or updates regarding the issue.
